jOOQ/src/main/java/org/jooq/impl/ParserImpl.java

```diff
@@ -1313,7 +1314,14 @@ private static final DDLQuery parseGrant(ParserContext ctx) {
 
         GrantOnStep s1 = privileges == null ? ctx.dsl.grant(privilege) : ctx.dsl.grant(privileges);
         GrantToStep s2 = s1.on(table);
-        return user == null ? s2.toPublic() : s2.to(user);
+        GrantWithGrantOptionStep s3 = user == null ? s2.toPublic() : s2.to(user);
+
+        if (peekKeyword(ctx, "WITH GRANT OPTION")) {
+            parseKeyword(ctx, "WITH GRANT OPTION");
```

Comment:
Notice that there's a `parseKeywordIf()` method, that does these two things together.

Reply:
I understand. Thank you!